wasm/jit/jit_arm64.go

```diff
@@ -303,56 +313,116 @@ func (c *arm64Compiler) emitPreamble() error {
 
-	return c.exit(jitCallStatusCodeReturned)
+	// Next we compare the decremented call frame stack pointer with the engine.precviousCallFrameStackPointer.
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineGlobalContextPreviouscallFrameStackPointer,
+		tmpReg,
+	)
+	c.applyTwoRegistersToNoneInstruction(arm64.ACMP, callFramePointerReg, tmpReg)
+
+	// If the values are identical, we return back to the Go code with returned status.
+	brIfNotEqual := c.newProg()
+	brIfNotEqual.As = arm64.ABNE
+	brIfNotEqual.To.Type = obj.TYPE_BRANCH
+	c.addInstruction(brIfNotEqual)
+	c.exit(jitCallStatusCodeReturned)
+
+	// Otherwise, we have to jump to the caller's return address.
+	c.setBRTargetOnNext(brIfNotEqual)
+
+	// First, we have to calculate the caller callFrame's absolute address to aquire the return address.
+	//
+	// "tmpReg = &engine.callFrameStack[0]"
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		reservedRegisterForEngine, engineGlobalContextCallFrameStackElement0AddressOffset,
+		tmpReg,
+	)
+	// "callFrameStackTopAddressRegister = tmpReg + callFramePointerReg << ${callFrameDataSizeMostSignificantSetBit}"
+	c.emitAddInstructionWithLeftShiftedRegister(
+		callFramePointerReg, callFrameDataSizeMostSignificantSetBit,
+		tmpReg,
+		callFrameStackTopAddressRegister,
+	)
+
+	// At this point, we have
+	//
+	//      [......., ra.caller, rb.caller, rc.caller, _, ra.current, rb.current, rc.current, _, ...]  <- call frame stack's data region (somewhere in the memory)
+	//                                                  |
+	//                               callFrameStackTopAddressRegister
+	//                   (absolute address of &callFrameStack[engine.callFrameStackPointer])
+	//
+	// where:
+	//      ra.* = callFrame.returnAddress
+	//      rb.* = callFrame.returnStackBasePointer
+	//      rc.* = callFrame.compiledFunction
+	//      _  = callFrame's padding (see comment on callFrame._ field.)
+	//
+	// What we have to do in the following is that
+	//   1) Set engine.valueStackContext.stackBasePointer to the value on "rb.caller".
+	//   2) Jump into the address of "ra.caller".
+
+	// 1) Set engine.valueStackContext.stackBasePointer to the value on "rb.caller".
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		// "rb.caller" is below the top address.
+		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnStackBasePointerOffset),
+		tmpReg)
+	c.applyRegisterToMemoryInstruction(arm64.AMOVD,
+		tmpReg,
+		reservedRegisterForEngine, engineValueStackContextStackBasePointerOffset)
+
+	// 2) Branch into the address of "ra.caller".
+	c.applyMemoryToRegisterInstruction(arm64.AMOVD,
+		// "rb.caller" is below the top address.
+		callFrameStackTopAddressRegister, -(callFrameDataSize - callFrameReturnAddressOffset),
+		tmpReg)
+	c.emitUnconditionalBranchToAddressOnRegister(tmpReg)
+
+	// They were temporarily used, so we mark them unused.
```

Comment:
when comments say the same thing as the english of the function calls, you an omit them.
```suggestion
```